finetrainers/models/hunyuan_video/control_specification.py

```diff
@@ -0,0 +1,358 @@
+        vae_dtype: torch.dtype = torch.bfloat16,
+        revision: Optional[str] = None,
+        cache_dir: Optional[str] = None,
+        condition_model_processors: List[ProcessorMixin] = None,
+        latent_model_processors: List[ProcessorMixin] = None,
+        control_model_processors: List[ProcessorMixin] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(
+            pretrained_model_name_or_path=pretrained_model_name_or_path,
+            tokenizer_id=tokenizer_id,
+            text_encoder_id=text_encoder_id,
+            transformer_id=transformer_id,
+            vae_id=vae_id,
+            text_encoder_dtype=text_encoder_dtype,
+            transformer_dtype=transformer_dtype,
+            vae_dtype=vae_dtype,
+            revision=revision,
+            cache_dir=cache_dir,
+        )
+
+        if condition_model_processors is None:
+            condition_model_processors = [
+                LlamaProcessor(["encoder_hidden_states", "encoder_attention_mask"]),
+                CLIPPooledProcessor(
+                    ["pooled_projections"],
+                    input_names={"tokenizer_2": "tokenizer", "text_encoder_2": "text_encoder"},
+                ),
+            ]
+        if latent_model_processors is None:
+            latent_model_processors = [HunyuanLatentEncodeProcessor(["latents", "latents_mean", "latents_std"])]
+        if control_model_processors is None:
+            control_model_processors = [HunyuanLatentEncodeProcessor(["control_latents", "__drop__", "__drop__"])]
+
+        self.condition_model_processors = condition_model_processors
+        self.latent_model_processors = latent_model_processors
+        self.control_model_processors = control_model_processors
+
+    @property
+    def control_injection_layer_name(self) -> str:
+        return "patch_embedding"
+
+    @property
+    def _resolution_dim_keys(self):
+        return {"latents": (2, 3, 4)}
+
+    load_condition_models = hunyuan_common.load_condition_models
+
+    load_latent_models = hunyuan_common.load_latent_models
+
+    load_pipeline = hunyuan_common.load_pipeline
+
+    def load_diffusion_models(self, new_in_features: int) -> Dict[str, torch.nn.Module]:
+        common_kwargs = {"revision": self.revision, "cache_dir": self.cache_dir}
+
+        if self.transformer_id is not None:
+            transformer = HunyuanVideoTransformer3DModel.from_pretrained(
+                self.transformer_id, torch_dtype=self.transformer_dtype, **common_kwargs
+            )
+        else:
+            transformer = HunyuanVideoTransformer3DModel.from_pretrained(
+                self.pretrained_model_name_or_path,
+                subfolder="transformer",
+                torch_dtype=self.transformer_dtype,
+                **common_kwargs,
+            )
+
+        # TODO(neph1): Need to update the patch embed layer
+        # transformer.x_embedder = HunyuanVideoPatchEmbed((patch_size_t, patch_size, patch_size), new_in_features)
```

Comment:
I've made some progress but could use some advice here. I believe I need to update this, but I'm not sure how. The line itself is taken from HunyuanVideoTransformer3DModel.

Or maybe I'm completely off track.